Sources/StreamVideo/WebRTC/v2/PeerConnection/MediaAdapters/LocalMediaAdapters/LocalVideoMediaAdapter.swift

```diff
@@ -285,50 +263,123 @@ final class LocalVideoMediaAdapter: LocalMediaAdapting, @unchecked Sendable {
     ///
     /// - Parameter activeEncodings: The set of active encoding identifiers.
     func changePublishQuality(
-        with activeEncodings: Set<String>
+        with layerSettings: [Stream_Video_Sfu_Event_VideoLayerSetting]
     ) {
-        Task { @MainActor [weak self] in
-            guard
-                let self,
-                let sender,
-                !activeEncodings.isEmpty
-            else {
-                return
-            }
+        guard
+            let sender,
+            !layerSettings.isEmpty
+        else {
+            return
+        }
 
-            var hasChanges = false
-            let params = sender
-                .sender
-                .parameters
-            var updatedEncodings = [RTCRtpEncodingParameters]()
+        var hasChanges = false
+        let params = sender
+            .sender
+            .parameters
 
-            for encoding in params.encodings {
-                guard let rid = encoding.rid else {
-                    continue
-                }
-                let shouldEnable = activeEncodings.contains(rid)
-
-                switch (shouldEnable, encoding.isActive) {
-                case (true, true):
-                    break
-                case (false, false):
-                    break
-                default:
-                    hasChanges = true
-                    encoding.isActive = shouldEnable
-                }
+        guard
+            !params.encodings.isEmpty
+        else {
+            log.warning("Update publish quality, No suitable video encoding quality found", subsystems: .webRTC)
+            return
+        }
+
+        let isUsingSVCCodec = {
+            if
+                let preferredCodec = params.codecs.first,
+                let videoCodec = VideoCodec(preferredCodec) {
+                return videoCodec.isSVC
+            } else {
+                return false
+            }
+        }()
+        var updatedEncodings = [RTCRtpEncodingParameters]()
+
+        for encoding in params.encodings {
+            let layerSettings = isUsingSVCCodec
+                // for SVC, we only have one layer (q) and often rid is omitted
+                ? layerSettings[0]
```

Comment:
this is a bit unsafe - can we use `first` instead?

Reply:
Yeap no issues on that. Worth mentioning that in the **current** code [0] is safe, as just above we have checked and ensured that the array isn't empty. As this is only a safety for the current code it makes absolute sense to be a little bit more abstract (and allow optionality) rather than risk having that being a bug on a future refactor.